test/pgo_trace_SUITE.erl

```diff
@@ -13,57 +11,63 @@ all() ->
                                              user => "test",
                                              password => "password",
                                              trace => true}),
-
+    _ = application:load(opentelemetry),
+    application:set_env(opentelemetry, processors, [{otel_simple_processor, #{}}]),
+    {ok, _} = application:ensure_all_started(opentelemetry),
     Config.
 
 end_per_suite(_Config) ->
     application:stop(pgo),
+    application:stop(opentelemetry),
+    application:unload(opentelemetry),
     ok.
 
 init_per_testcase(_, Config) ->
-    _ = application:load(opentelemetry),
-    application:set_env(opentelemetry, processors, [{otel_batch_processor, #{scheduled_delay_ms => 1}}]),
-    {ok, _} = application:ensure_all_started(opentelemetry),
-    otel_batch_processor:set_exporter(otel_exporter_pid, self()),
+    otel_simple_processor:set_exporter(otel_exporter_pid, self()),
     Config.
 
 end_per_testcase(_, _Config) ->
-    _ = application:stop(opentelemetry),
+    otel_simple_processor:set_exporter(none),
     ok.
 
+-define(assertReceive(SpanName),
+        receive
+            {span, Span=#span{name=SpanName}} ->
+                Span
+        after
+            1000 ->
+                ct:fail("Did not receive the span after 1s")
+        end).
+
 trace_query(_Config) ->
     ?assertMatch(#{rows := [{empty}]},
                  pgo:query("select '[1,1)'::int4range", [], #{})),
-
-    receive
-        {span, #span{name=Name,
-                     parent_span_id=Parent,
-                     attributes=Attributes,
-                     events=_TimeEvents}} when Parent =:= undefined ->
-            ?assertEqual(<<"pgo:query/3">>, Name),
-            ?assertMatch(#{<<"db.system">> := <<"postgresql">>,
-                           <<"db.name">> := <<"test">>,
-                           <<"db.statement">> := <<"select '[1,1)'::int4range">>,
-                           <<"db.user">> := <<"test">>}, otel_attributes:map(Attributes))
-    after
-        5000 ->
-            ct:fail(timeout)
-    end,
+    #span{attributes={attributes, 128, infinity, 0, RecievedAttributes}} = 
+        ?assertReceive(<<"pgo:query/3">>),
+    #{<<"db.name">> := DbName,
+      <<"db.statement">> := DbStatement,
+      <<"db.system">> := DbSystem,
+      <<"db.user">> := DbUser} = RecievedAttributes,
+    ?assertMatch(DbName, <<"test">>),
+    ?assertMatch(DbStatement, <<"select '[1,1)'::int4range">>),
+    ?assertMatch(DbSystem, <<"postgresql">>),
+    ?assertMatch(DbUser, <<"test">>),
     ok.
 
 trace_with_parent_query(_Config) ->
     SpanCtx = ?start_span(<<"parent-of-query">>),
     ?set_current_span(SpanCtx),
 
     ?assertMatch(#{rows := [_]},
-                 pgo:query("select '[1,2)'::int4range", [], #{trace => true})),
+                 % TODO: Some decoding issue here when this is '[1, 2)'
+                 pgo:query("select '[1,1)'::int4range", [], #{trace => true})),
```

Comment:
No idea why this is failing but I get 
```
{pg_range,decode_range,
         [2,<<0,0,0,4,0,0,0,1,0,0,0,4,0,0,0,2>>,unknown_oid],
         []},
     {pgo_protocol,decode_row0,5,
         [{file,"/home/barry/programming/erlang/pgo/src/pgo_protocol.erl"},
          {line,591}]},
...
```

Reply:
The OID in pgo_internal.hrl looks right according to https://github.com/postgres/postgres/blob/4effd0844daf41a962c05188b719df75ca8ced93/src/include/catalog/pg_type.dat#L468-L472

Reply:
Alright, I don't fully understand what `assertMatch` means if you ignore stuff and it turns out this was failing. So, I just put in the return values and it is fine.